.github/workflows/publish-pr-preview.yml

```diff
@@ -37,13 +39,15 @@ jobs:
         with:
           surge-token: ${{ secrets.SURGE_TOKEN_DOC }}
           github-token: ${{ secrets.GITHUB_TOKEN }}
-          doc-site-branch: ${{ env.BRANCH_NAME }}
+          doc-site-branch: ${{ github.head_ref }}
```

Comment:
the bash script that compute the `BRANCH_NAME` env var should be removed as the variable is not used anymore